RiotSwiftUI/Modules/UserSessions/UserSessionsOverview/Service/MatrixSDK/UserSessionsOverviewService.swift

```diff
@@ -15,11 +15,123 @@
 //
 
 import Foundation
+import MatrixSDK
 
 class UserSessionsOverviewService: UserSessionsOverviewServiceProtocol {
     
+    // MARK: - Constants
+    
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    private let mxSession: MXSession
+    
+    // MARK: Public
+        
+    private(set) var lastOverviewData: UserSessionsOverviewData
+    
     // MARK: - Setup
     
-    init() {
+    init(mxSession: MXSession) {
+        self.mxSession = mxSession
+        
+        self.lastOverviewData =  UserSessionsOverviewData(currentSessionInfo: nil, unverifiedSessionsInfo: [], inactiveSessionsInfo: [], otherSessionsInfo: [])
```

Comment:
Looks like this is overwritten by `setupInitialOverviewData` right below?

Reply:
Yes in fact the annoying part is that the view state property is not nullable and should be initialized 
before calling `self`. So we should call `super.init(initialViewState: initialViewState)` before calling `self`. And it means otherwise we need to make static methods or intermediate classes just for this case :/

Reply:
Ohh, it's a `self` issue. Ok, got it. Seems ok then.